packages/cli-kit/src/public/node/notifications-system.ts

```diff
@@ -0,0 +1,235 @@
+import {versionSatisfies} from './node-package-manager.js'
+import {renderError, renderInfo, renderWarning} from './ui.js'
+import {getCurrentCommandId} from './global-context.js'
+import {fileExists, readFile} from './fs.js'
+import {outputDebug} from './output.js'
+import {CLI_KIT_VERSION} from '../common/version.js'
+import {NotificationsKey, cacheRetrieveOrRepopulate, getCache, setCache} from '../../private/node/conf-store.js'
+
+const URL = 'https://raw.githubusercontent.com/Shopify/cli/notifications/notifications.json'
+
+export interface Notifications {
+  notifications: Notification[]
+}
+
+export interface Notification {
+  id: string
+  message: string
+  cta?: {
+    label: string
+    url: string
+  }
+  type: 'info' | 'warning' | 'error'
+  title?: string
+  minVersion?: string
+  maxVersion?: string
+  minDate?: string
+  maxDate?: string
+  commands?: string[]
+  surface?: 'app' | 'theme' | 'hydrogen' | string
+  frequency?: 'always' | 'once' | 'once_a_day' | 'once_a_week'
+}
+
+/**
+ * Shows notifications to the user if they meet the criteria specified in the notifications.json file.
+ *
+ * @param currentSurfaces - The surfaces present in the current project (usually for app extensions).
+ * @returns - A promise that resolves when the notifications have been shown.
+ */
+export async function showNotificationsIfNeeded(currentSurfaces?: string[]): Promise<void> {
+  try {
+    const notifications = await getNotifications()
+    const commandId = getCurrentCommandId()
+    const notificationsToShow = filterNotifications(notifications.notifications, commandId, currentSurfaces)
+    await renderNotifications(notificationsToShow)
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch (error: unknown) {
+    outputDebug('Error retrieving notifications')
+  }
+}
+
+/**
+ * Renders the notifications to the user.
+ *
+ * @param notifications - The notifications to render.
+ */
+async function renderNotifications(notifications: Notification[]) {
+  notifications.forEach((notification) => {
```

Comment:
I think this is going to be quite noisy if there are multiple notifications matching on any given command run -- each section having it's own box. I'm not sure what the best alternative is. Or, how likely it is to happen. What do you think?

Reply:
Limited to 2 messages finally